trade-module/src/main/java/io/edpn/backend/trade/application/dto/v1/LocateCommodityRequest.java

```diff
@@ -1,27 +1,29 @@
 package io.edpn.backend.trade.application.dto.v1;
 
 import io.swagger.v3.oas.annotations.media.Schema;
-import jakarta.validation.Valid;
 import jakarta.validation.constraints.NotNull;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.Setter;
 import lombok.extern.jackson.Jacksonized;
 
-import java.util.UUID;
-
 @Getter
+@Setter
```

Comment:
Should we have Setter on a Request object? What's the use-case for overwriting parts of the incoming request?

Reply:
It is an object that is mapped from json to pojo, so it doesn't hurt.

an `@AllArgsConstructor` and final properties might be better indeed

Reply:
Amended. Code seems to compile and run correctly is @Setter is removed. I can pull it.

Reply:
It's a non-functional issue, it just struck my brain as being odd. AllArgs/RequiredArgs should allow the object to be created.

But I've also just hit the approval button, so meh.

Reply:
It doesn't work. If you could confirm. Delete @Setter and then run /commodity/filter endpoint and see if it returns an error.